packages/algob/src/lib/tx.ts

```diff
@@ -149,31 +186,80 @@ async function sendAndWait (
 export async function executeTransaction (
   deployer: Deployer,
   execParams: rtypes.ExecParams | rtypes.ExecParams[]): Promise<algosdk.ConfirmedTxInfo> {
-  const suggestedParams = await getSuggestedParams(deployer.algodClient);
-  const mkTx = async (p: rtypes.ExecParams): Promise<Transaction> =>
-    mkTransaction(p,
-      await mkTxParams(deployer.algodClient, p.payFlags, Object.assign({}, suggestedParams)));
-
-  let signedTxn;
-  if (Array.isArray(execParams)) {
-    if (execParams.length > 16) { throw new Error("Maximum size of an atomic transfer group is 16"); }
-
-    let txns = [];
-    for (const t of execParams) { txns.push(await mkTx(t)); }
-    txns = algosdk.assignGroupID(txns);
-    signedTxn = txns.map((txn: Transaction, index: number) => {
-      const signed = signTransaction(txn, execParams[index]);
-      deployer.log(`Signed transaction ${index}`, signed);
-      return signed;
-    });
-  } else {
-    const txn = await mkTx(execParams);
-    signedTxn = signTransaction(txn, execParams);
-    deployer.log(`Signed transaction:`, signedTxn);
+  try {
+    let signedTxn;
+    let txns: Transaction[] = [];
+    const txIDMap = new Map<number, string>();
+    if (Array.isArray(execParams)) {
+      if (execParams.length > 16) { throw new Error("Maximum size of an atomic transfer group is 16"); }
+
+      for (const [index, t] of execParams.entries()) {
```

Comment:
Let's use `[i, txn]`/`[i, tx]` like you did in line 235

Reply:
[idx, txn]